pkg/interface/chat/src/js/api.js

```diff
@@ -77,6 +77,16 @@ class UrbitApi {
     });
   }
 
+  addReadReceiptPending(station) {
+    const m = store.state.myReadReceiptsPending.set(station, true);
+    store.setState({myReadReceiptsPending: m})
+  }
+
+  removeReadReceiptPending(station) {
+    const m = store.state.myReadReceiptsPending.set(station, false);
+    store.setState({myReadReceiptsPending: m})
```

Comment:
Both of these probably shouldn't directly manipulate the state from api.js — I would make a fake `chat-update` object and throw it to `store.handleEvent` like the [sidebarToggle does](https://github.com/urbit/urbit/blob/9751bb9daadac160300ffa313be511803b348e79/pkg/interface/chat/src/js/api.js#L185). We want state manipulation to happen in one place (the reducers), basically.

(I know I wrote [the other function that has this behaviour](https://github.com/urbit/urbit/blob/9751bb9daadac160300ffa313be511803b348e79/pkg/interface/chat/src/js/api.js#L68), but I know better now. Probably, both should be changed...)